File: solution/210824/가장큰수.py
def solution(numbers):
    answer = ''

    sort = quick_sort(numbers)
    if(sort[0]=='0'):
        return '0'

    for s in sort:
        answer+=s

    return answer


def quick_sort(_numbers):
    if len(_numbers) <= 1:
        return _numbers
    pivot = str(_numbers[0])

    greater_arr, lesser_arr, equal_arr = [], [], []

    for num in _numbers:
        if int(str(num) + pivot) > int(pivot + str(num)):
            greater_arr.append(str(num))
        elif int(str(num) + pivot) < int(pivot + str(num)):
            lesser_arr.append(str(num))
        else:
            equal_arr.append(str(num))

    return quick_sort(greater_arr) + equal_arr + quick_sort(lesser_arr)

File: solution/211001/가장큰수.java
class Solution {
    public void sort(int[] data, int l, int r){
        int left = l;
        int right = r;
        int pivot = data[(l+r)/2];

        do{
            while(Long.parseLong(Integer.toString(data[left])+Integer.toString(pivot)) > Long.parseLong(Integer.toString(pivot)+Integer.toString(data[left])) ) left++;
            while(Long.parseLong(Integer.toString(data[right])+Integer.toString(pivot)) < Long.parseLong(Integer.toString(pivot)+Integer.toString(data[right])) ) right--;

            if(left <= right){
                int temp = data[left];
                data[left] = data[right];
                data[right] = temp;
                left++;
                right--;
            }
        }while (left <= right);

        if(l < right) sort(data, l, right);
        if(r > left) sort(data, left, r);
    }

    public String solution(int[] numbers){
        sort(numbers,0, numbers.length-1);
        String answer = "";
        if (numbers[0] == 0){
            return "0";
        }
        for (int number : numbers) {
            answer += Integer.toString(number);
        }
        return answer;
    }

}
